src/main/java/com/epam/spring/homework3/repository/impl/DishRepositoryImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.epam.spring.homework3.repository.impl;
+
+import com.epam.spring.homework3.model.Dish;
+import com.epam.spring.homework3.repository.DishRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+@Slf4j
+@Component
+public class DishRepositoryImpl implements DishRepository {
+
+    private final List<Dish> list = new ArrayList<>();
+
+    @Override
+    public List<Dish> findAll() {
+        log.info("find all dishes");
+        return new ArrayList<>(list);
+    }
+
+    @Override
+    public Dish findById(Long id) {
+        log.info("find dish with id {}", id);
+        return list.stream()
+                .filter(dish -> dish.getId().equals(id))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Dish is not found!"));
```

Comment:
I would recommend to create specific custom exception, like DishNotFoundException instead of abstract RuntimeException